tooling/bin/assess-minibroker-charts.rb

```diff
@@ -0,0 +1,622 @@
+  @manifest = File.join(appdir, "manifest.yml") unless @manifest
+  @manifest
+end
+
+def chart_index
+  @chartindex = File.join(appdir, "index.yaml") unless @chartindex
+  @chartindex
+end
+
+def archive_app
+  @aapp = File.join(appdir, chart_in_app) unless @aapp
+  @aapp
+end
+
+def domain
+  @domain, _, _ = capture "kubectl get pods -o json --namespace \"#{@namespace}\" api-group-0 | jq -r '.spec.containers[0].env[] | select(.name == \"DOMAIN\").value'" unless @domain
+  @domain
+end
+
+def tester
+  @brain = File.join(@top, "make/tests") unless @brain
+  @brain
+end
+
+def appsrc
+  @appsrc = File.join(@top, "src/scf-release/src/acceptance-tests-brain/test-resources/node-env") unless @appsrc
+  @appsrc
+end
+
+def helm_repo
+  @helmrepo = "http://#{helm_app}.#{domain}" unless @helmrepo
+  @helmrepo
+end
+
+def target
+  @target = "https://api.#{domain}" unless @target
+  @target
+end
+
+def chart_ref
+  @chartref = "#{helm_repo}/#{chart_in_app}" unless @chartref
+  @chartref
+end
+
+# ......................................................................
+# Logging, and terminal cursor control
+# First three commands snarfed from test utils and modified to suit
+# (divert output into log file)
+
+# Set global options.
+# If a block is given, the options are only active in that block.
+def set(opts={})
+    if block_given?
+        old_opts = $opts.dup
+        $opts.merge! opts
+        yield
+        $opts.merge! old_opts
+    else
+        $opts.merge! opts
+    end
+end
+
+def run(*args)
+  opts = $opts.dup
+  opts.merge! args.last if args.last.is_a? Hash
+  _, _, status = capture(*args)
+  return unless opts[:errexit]
+  unless status.success?
+    # Print an error at the failure site
+    puts "Command exited with #{status.exitstatus}".red
+    fail "Command exited with #{status.exitstatus}"
+  end
+end
+
+def capture(*args)
+  _print_command(*args)
+  args.last.delete :errexit if args.last.is_a? Hash
+  stdout, stderr, status = Open3.capture3(*args)
+  log stdout
+  log stderr.red
+  return stdout.chomp, stderr.chomp, status
+end
+
+# Internal helper: print a command line in the log.
+def _print_command(*args)
+    cmd = args.dup
+    cmd.shift if cmd.first.is_a? Hash
+    cmd.pop if cmd.last.is_a? Hash
+    log "+ #{cmd.join(" ")}".bold
+    log "\n"
+end
+
+def separator(text)
+  sepline = "____________________________________________________________ #{text} ___"
+  log "\n#{sepline.magenta}\n"
+  write text
+  message = yield
+  left text.length
+  eeol
+  write message
+end
+
+def log_start(engine,enginev,chartv)
+  enginedir = File.join(@workdir, engine)
+  @log = File.join(enginedir, "#{enginev}-#{chartv}.log")
+  FileUtils.mkdir_p(enginedir)
+  FileUtils.rm_f(@log)
+  FileUtils.touch(@log)
+end
+
+def log(text)
+  # append to log file @log
+  # https://stackoverflow.com/questions/27956638/how-to-append-a-text-to-file-succinctly/27956730
+  #File.write(@log, text, File.size(@log), mode: 'a')
+  File.write(@log, text, mode: 'a')
```

Comment:
Or just have `@log` be an opened `File` (with `@log.sync = true`), so we don't need to re-open the file every time.

```ruby
def log_start(engine, enginev, chartv)
  enginedir = File.join(@workdir, engine)
  FileUtils.mkdir_p(enginedir)
  @log = File.open(File.join(enginedir, "#{enginev}-#{chartv}.log"), File::CREAT|File::TRUNC|File::APPEND)
end

def log(text)
  @log.write text # not puts?
end
```

Reply:
`write` was copy from the SO. `puts` might work as well. done local.